app/utils/rentry/wrapper.py

```diff
@@ -1,93 +1,98 @@
+    def get_csrf_token(self):
+        # Get CSRF token from the response cookies after making a GET request to the base URL
+        response = self.get(BASE_URL)
+        return response.cookies.get("csrftoken")
 
+class RentryUpload:
     def __init__(self, text: str):
-        response = self.new(text)
-        if response["status"] != "200":
-            self.upload_success = False
-            self.url = None
-            logger.error("error: {}".format(response["content"]))
-            try:
-                for i in response["errors"].split("."):
-                    i and logger.warning(i)
-                logger.error("RentryUpload failed!")
-            except:
-                logger.error("RentryUpload failed!")
-        else:
-            self.upload_success = True
-            self.url = response["url"]
-            logger.debug("RentryUpload successfully uploaded data!")
-            logger.debug("Url: {}".format(self.url))
-            logger.debug("Edit code: {}".format(response["edit_code"]))
+        self.upload_success = False
+        self.url = None
+
+        try:
+            response = self.new(text)
+            if response.get("status") != "200":
+                self.handle_upload_failure(response)
+            else:
+                self.upload_success = True
+                self.url = response["url"]
+        finally:
+            if self.upload_success:
+                logger.debug(
+                    f"RentryUpload successfully uploaded data! Url: {self.url}, Edit code: {response['edit_code']}")
+
+    def handle_upload_failure(self, response):
+        error_content = response.get("content", "Unknown")
+        errors = response.get("errors", "").split(".")
+        logger.error(f"Error: {error_content}")
+        for error in errors:
+            error and logger.warning(error)
+        logger.error("RentryUpload failed!")
 
     def new(self, text):
-        client, cookie = UrllibClient(), SimpleCookie()
+        # Initialize an UrllibClient for making HTTP requests
+        client = UrllibClient()
 
-        cookie.load(vars(client.get("https://rentry.co"))["headers"]["Set-Cookie"])
-        csrftoken = cookie["csrftoken"].value
+        # Get CSRF token for authentication
+        csrf_token = client.get_csrf_token()
 
+        # Prepare payload for the POST request
         payload = {
-            "csrfmiddlewaretoken": csrftoken,
+            "csrfmiddlewaretoken": csrf_token,
             "text": text,
         }
 
-        return json_loads(
-            client.post("https://rentry.co/api/new", payload, headers=_headers).data
-        )
+        # Perform the POST request to create a new entry
+        return json_loads(client.post(API_NEW_ENDPOINT, payload, headers=_HEADERS).data)
 
 
 class RentryImport(QDialog):
     def __init__(self):
         super().__init__()
-        self.package_ids: list[str] = []  # Initialize an empty list to store package_ids
```

Comment:
Why remove typing?

Reply:
got rid of list[str] being and extra step and importing modlist without it works as well can add it back if its
so now its
self.package_ids = []
before it was 
self.package_ids: list[str] = []